internal/cloudapi/server.go

```diff
@@ -0,0 +1,222 @@
+	}
+	imageRequests := make([]imageRequest, len(request.ImageRequests))
+	var targets []*target.Target
+
+	for i, ir := range request.ImageRequests {
+		arch, err := distribution.GetArch(ir.Architecture)
+		if err != nil {
+			http.Error(w, fmt.Sprintf("Unsupported architecture '%s' for distribution '%s'", ir.Architecture, request.Distribution), http.StatusBadRequest)
+			return
+		}
+		imageType, err := arch.GetImageType(ir.ImageType)
+		if err != nil {
+			http.Error(w, fmt.Sprintf("Unsupported image type '%s' for %s/%s", ir.ImageType, ir.Architecture, request.Distribution), http.StatusBadRequest)
+			return
+		}
+		repositories := make([]rpmmd.RepoConfig, len(ir.Repositories))
+		for j, repo := range ir.Repositories {
+			repositories[j].BaseURL = repo.Baseurl
+			repositories[j].RHSM = true
+		}
+
+		var bp = blueprint.Blueprint{}
+		err = bp.Initialize()
+		if err != nil {
+			http.Error(w, "Unable to initialize blueprint", http.StatusInternalServerError)
+			return
+		}
+
+		packageSpecs, _ := imageType.Packages(bp)
+		packages, _, err := server.rpmMetadata.Depsolve(packageSpecs, nil, repositories, distribution.ModulePlatformID(), arch.Name())
+		if err != nil {
+			http.Error(w, fmt.Sprintf("Failed to depsolve base base packages for %s/%s/%s: %s", ir.ImageType, ir.Architecture, request.Distribution, err), http.StatusBadRequest)
+			return
+		}
+		buildPackageSpecs := imageType.BuildPackages()
+		buildPackages, _, err := server.rpmMetadata.Depsolve(buildPackageSpecs, nil, repositories, distribution.ModulePlatformID(), arch.Name())
+		if err != nil {
+			http.Error(w, fmt.Sprintf("Failed to depsolve build packages for %s/%s/%s: %s", ir.ImageType, ir.Architecture, request.Distribution, err), http.StatusBadRequest)
+			return
+		}
+
+		imageOptions := distro.ImageOptions{Size: imageType.Size(0)}
+		if request.Customizations != nil && request.Customizations.Subscription != nil {
+			imageOptions.Subscription = &distro.SubscriptionImageOptions{
+				Organization:  request.Customizations.Subscription.Organization,
+				ActivationKey: request.Customizations.Subscription.ActivationKey,
+				ServerUrl:     request.Customizations.Subscription.ServerUrl,
+				BaseUrl:       request.Customizations.Subscription.BaseUrl,
+				Insights:      request.Customizations.Subscription.Insights,
+			}
+		}
+
+		manifest, err := imageType.Manifest(nil, imageOptions, repositories, packages, buildPackages)
+		if err != nil {
+			http.Error(w, fmt.Sprintf("Failed to get manifest for for %s/%s/%s: %s", ir.ImageType, ir.Architecture, request.Distribution, err), http.StatusBadRequest)
+			return
+		}
+
+		imageRequests[i].manifest = manifest
+
+		if len(*ir.UploadRequests) != 1 {
+			http.Error(w, "Only compose requests with a single upload target are currently supported", http.StatusBadRequest)
+			return
+		}
+		uploadRequest := (*ir.UploadRequests)[0]
+		/* oneOf is not supported by the openapi generator so marshal and unmarshal the uploadrequest based on the type */
```

Comment:
Is there a way to get the generator to force `json.RawMessage` fields here, so that we don't have to go through this dance?

Reply:
I'm not sure what you mean here, do you mean making the (aws)uploadrequests a json object without any further specification?

Reply:
Yeah, that's how you'd do it manually (`json.RawMessage` is just `[]bye`, so you can defer unmarshalling until you know the type). I haven't looked into how we get the generator to do that, though. Might not be possible.

Reply:
so I'm trying this out but i'm not sure, I can make upload_requests an array of type 'object' and get a `[]map[string]interface{}`.

 However with the dance we can define all the expected types and what they should look like in the yml.

Reply:
Yeah now that I've looked into the generator a bit more, I realized as well that it cannot handle this (other than `map[string]interface{}`, which is useless). Feel free to ignore me!